frontend/rust-lib/flowy-database2/src/services/field/type_options/selection_type_option/multi_select_type_option.rs

```diff
@@ -140,31 +141,51 @@ impl TypeOptionCellDataCompare for MultiSelectTypeOption {
     &self,
     cell_data: &<Self as TypeOption>::CellData,
     other_cell_data: &<Self as TypeOption>::CellData,
+    sort_condition: SortCondition,
   ) -> Ordering {
-    for i in 0..min(cell_data.len(), other_cell_data.len()) {
-      let order = match (
-        cell_data
-          .get(i)
-          .and_then(|id| self.options.iter().find(|option| &option.id == id)),
-        other_cell_data
-          .get(i)
-          .and_then(|id| self.options.iter().find(|option| &option.id == id)),
-      ) {
-        (Some(left), Some(right)) => left.name.cmp(&right.name),
-        (Some(_), None) => Ordering::Greater,
-        (None, Some(_)) => Ordering::Less,
-        (None, None) => default_order(),
-      };
-
-      if order.is_ne() {
-        return order;
-      }
+    match cell_data.len().cmp(&other_cell_data.len()) {
+      Ordering::Equal => {
+        for (left_id, right_id) in cell_data.iter().zip(other_cell_data.iter()) {
+          let left = self.options.iter().find(|option| &option.id == left_id);
+          let right = self.options.iter().find(|option| &option.id == right_id);
+
+          let order = match (left, right) {
+            (None, None) => Ordering::Equal,
+            (None, Some(_)) => Ordering::Greater,
+            (Some(_), None) => Ordering::Less,
```

Comment:
If the left side has a value, then the comparison would be: (Some(_), None) => Ordering::Greater

Reply:
Oh. I wanted to move the non-empty cells to the front and the empty ones to the back. That's why I write Ordering::Less

Reply:
Got it. It would be better to have documentation there.